### ISSUE TITLE
expo publish never finishes

### ISSUE THREAD

**Issue by TMcLoone** [expo/expo-cli#3021]
## Description

Expo Publish never finishing publishing. 

## Environment

  Expo CLI 4.0.13 environment info:
    System:
      OS: macOS 11.0.1
      Shell: 5.8 - /bin/zsh
    Binaries:
      Node: 14.3.0 - ~/.nvm/versions/node/v14.3.0/bin/node
      Yarn: 1.22.4 - /usr/local/bin/yarn
      npm: 6.14.9 - ~/.nvm/versions/node/v14.3.0/bin/npm
      Watchman: 4.9.0 - /usr/local/bin/watchman
    Managers:
      CocoaPods: 1.10.0 - /usr/local/bin/pod
    SDKs:
      iOS SDK:
        Platforms: iOS 14.2, DriverKit 20.0, macOS 11.0, tvOS 14.2, watchOS 7.1
    IDEs:
      Android Studio: 4.1 AI-201.8743.12.41.6953283
      Xcode: 12.2/12B45b - /usr/bin/xcodebuild
    npmPackages:
      expo: ^40.0.0 => 40.0.0
      react: 16.13.1 => 16.13.1
      react-dom: 16.13.1 => 16.13.1
      react-native: 0.63.4 => 0.63.4
    npmGlobalPackages:
      expo-cli: 4.0.13
    Expo Workflow: bare

## Expected Behavior

Expo publish finishes and update is available

## Observed Behavior

It gets to 'publish complete' with a link to manifest but hangs after that. It appears that the updates don't become available when trying to update from within the app.

**Comment by brentvatne** [comment]
can you please explain in more detail? and share logs? we need more information to be able to help you

**Comment by TMcLoone** [comment]
Running 

expo publish --release-channel production-1-0-1

I get to the point in the screenshot below and it just hangs, I end up having to abort the process. Seems possibly related to upgrading to expo 40 yesterday.

![CleanShot 2020-12-11 at 12 44 38@2x](https://user-images.githubusercontent.com/2749978/101942188-c0355100-3bae-11eb-84e4-adb923db581e.jpg)

**Comment by brentvatne** [comment]
this could be related to this from release notes: "@expo/dev-server is used in all apps SDK 40 and onward"

you could try this suggestion:

> If you run into any issues in SDK 40 and suspect this change may be related, you can revert back to using the old process by setting the environment variable EXPO_USE_DEV_SERVER=0.

**Comment by TMcLoone** [comment]
No luck unfortunately

**Comment by TMcLoone** [comment]
After some more testing, I found that it does actually upload and publish, just the process never exits. I rolled back to older versions of the CLI and it does the same so guessing it has something to do with expo 40

**Comment by brentvatne** [comment]
@TMcLoone - why would it have something to do with sdk 40 if rolling back your expo CLI version fixes it? did you also roll back the SDK version? can you share your project with me on github?

**Comment by TMcLoone** [comment]
@brentvatne  Rolling back does not fix it. I did not try rolling back the SDK, the project is in production so unable to share. I'll try to see if I can re-create it in another project.

**Comment by brentvatne** [comment]
one thing that is strange from the output you shared is that the "Welcome to React Native" message shows up. this should not be happening. here is an example of me running `expo publish` on a bare workflow project on sdk 40:

![Screen Shot 2020-12-12 at 5 07 20 PM](https://user-images.githubusercontent.com/90494/102000455-7fb40100-3c9c-11eb-8c8d-367dca4bb771.png)

**Comment by brentvatne** [comment]
i suspect that whatever is causing that to start is the source of the problem. i imagine that it's not being terminated properly and so the command is hanging. without more information on your project i can't help, as i don't know what would be causing that on your end. so i will close this issue until there is a reproducible example or sufficient information to re-create those circumstances

**Comment by brkastner** [comment]
I ran into this issue trying to merge `@ui-kitten/metro-config` with expo to bootstrap a custom theme mapping. After digging into their code, it looks like they open a file watcher on `mapping.json` that prevents the server from terminating at the end. 

Here's the relevant code from their merge config utility:
https://github.com/akveo/react-native-ui-kitten/blob/master/src/metro-config/index.ts#L60-L64

Here's the open `@ui-kitten/metro-config`  issue where I posted my specific workaround:
https://github.com/akveo/react-native-ui-kitten/issues/1410

@brentvatne this issue should probably be reopened if there is a way for Expo to forcefully close any opened file watchers on finish since it should resolve this issue. Sadly I know very little about Metro or this process as a whole otherwise I'd take a stab at making a PR myself.
